src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java

```
@@ -341,4 +342,45 @@ public static void validateArraySchema(Schema arraySchema, String name) {
+                                                       columnName));
+    }
+    for (Schema.Field field : outputSchema.getFields()) {
+      if (field.getName().equals(columnName)) {
+        if (!field.getSchema().getType().equals(Schema.Type.UNION)) {
+          if (field.getSchema().getLogicalType() != null
+            && (field.getSchema().getLogicalType().equals(Schema.LogicalType.DATE)
+            || field.getSchema().getLogicalType().equals(Schema.LogicalType.TIMESTAMP_MICROS))) {
+            return;
+          }
+        } else {
+          for (Schema schema : field.getSchema().getUnionSchemas()) {
+            if (!schema.getType().equals(Schema.Type.NULL)
+              && (Schema.LogicalType.DATE.equals(schema.getLogicalType())
+              || Schema.LogicalType.TIMESTAMP_MICROS.equals(schema.getLogicalType()))) {
+              return;
+            }
+          }
+        }
+        Schema.Type type = !Schema.Type.UNION.equals(field.getSchema().getType())
+          ? field.getSchema().getType()
+          : field.getSchema().getUnionSchema(0).getType();
+        Schema.LogicalType fieldType = !Schema.Type.UNION.equals(field.getSchema().getType())
+          ? field.getSchema().getLogicalType()
+          : field.getSchema().getUnionSchema(0).getLogicalType();
+        throw new InvalidStageException(String.format("Column '%s' is of type '%s' (logical type '%s'), " +
+                                                           "expected logical type is '%s' or '%s'",
+                                                      columnName, type, fieldType,
+                                                      Schema.LogicalType.DATE,
+                                                      Schema.LogicalType.TIMESTAMP_MICROS));
+      }
+    }
+    throw new InvalidStageException(String.format("Column '%s' missing in output table schema", columnName));
```

This should throw an InvalidConfigPropertyException where it sets the partition column property as the invalid property, with message:

Partition column '%s' is missing from the table schema.

---

Done.